Incorrect ASSERT format in scatter op

We don't support printed asserts in kernels. This gets hit in the DeepSeek demo with watcher on resulting in compile failures.

```
[1,0]<stdout>: 2025-11-27 21:31:43.429 | error    |    BuildKernels | ncrisc compile failure -- cmd: cd /home/sjameel/.cache_g05glx03_rank0/tt-metal-cache/434580b8a4/2466170624573553914/kernels/reader_scatter/4372489414908321291/ncrisc/ && /home/sjameel/tt-metal/runtime/sfpi/compiler/bin/riscv-tt-elf-g++ -O2 -std=c++17 -flto=auto -ffast-math -fno-exceptions -g -MMD -fno-use-cxa-atexit -Wall -Werror -Wno-unknown-pragmas -Wno-deprecated-declarations -Wno-error=multistatement-macros -Wno-error=parentheses -Wno-error=unused-but-set-variable -Wno-unused-variable -Wno-unused-function -mcpu=tt-wh -fno-tree-loop-distribute-patterns -I. -I.. -I/home/sjameel/tt-metal/ -I/home/sjameel/tt-metal/ttnn -I/home/sjameel/tt-metal/ttnn/cpp -I/home/sjameel/tt-metal/tt_metal -I/home/sjameel/tt-metal/tt_metal/include -I/home/sjameel/tt-metal/tt_metal/hw/inc -I/home/sjameel/tt-metal/tt_metal/hostdevcommon/api -I/home/sjameel/tt-metal/tt_metal/hw/inc/debug -I/home/sjameel/tt-metal/tt_metal/api/ -I/home/sjameel/tt-metal/tt_metal/api/tt-metalium/ -I/home/sjameel/tt-metal/tt_metal/hw/ckernels/wormhole_b0/metal/common -I/home/sjameel/tt-metal/tt_metal/hw/ckernels/wormhole_b0/metal/llk_io -I/home/sjameel/tt-metal/tt_metal/hw/inc/tt-1xx -I/home/sjameel/tt-metal/tt_metal/hw/inc/tt-1xx/wormhole -I/home/sjameel/tt-metal/tt_metal/hw/inc/tt-1xx/wormhole/wormhole_b0_defines -I/home/sjameel/tt-metal/tt_metal/hw/inc/tt-1xx/wormhole/noc -I/home/sjameel/tt-metal/tt_metal/lite_fabric/hw/inc/wormhole -I/home/sjameel/tt-metal/tt_metal/third_party/tt_llk/tt_llk_wormhole_b0/common/inc -I/home/sjameel/tt-metal/tt_metal/third_party/tt_llk/tt_llk_wormhole_b0/llk_lib -I/home/sjameel/tt-metal/tt_metal/hw/firmware/src/tt-1xx -c -o ncrisck.o /home/sjameel/tt-metal/tt_metal/hw/firmware/src/tt-1xx/ncrisck.cc -DIS_NOT_POW2_NUM_DRAM_BANKS=1 -DIS_NOT_POW2_NUM_L1_BANKS=1 -DNUM_DRAM_BANKS=12 -DNUM_L1_BANKS=72 -DPCIE_NOC_X=0 -DPCIE_NOC_Y=3 -DTENSIX_FIRMWARE -DLOCAL_MEM_EN=0 -DWATCHER_ENABLED -DPROCESSOR_INDEX=1 -DCOMPILE_FOR_NCRISC -DARCH_WORMHOLE -DDISPATCH_MESSAGE_ADDR=4290183304 -DKERNEL_BUILD -DROUTING_MODE='137' -DNOC_INDEX='0' -DNOC_MODE='0' -DKERNEL_COMPILE_TIME_ARGS=70920832,70925184,70920928,70925280,0,2,1,3,8,4,4,8,16,8,8,16,4,2,2,2,2  (build.cpp:60)
[1,0]<stdout>: 2025-11-27 21:31:43.429 | critical |          Always | ncrisc build failed. Log: In file included from /home/sjameel/tt-metal/tt_metal/hw/inc/dataflow_api_addrgen.h:9,
[1,0]<stdout>:                  from /home/sjameel/tt-metal/tt_metal/hw/inc/dataflow_api.h:20,
[1,0]<stdout>:                  from /home/sjameel/tt-metal/tt_metal/hw/inc/firmware_common.h:20,
[1,0]<stdout>:                  from /home/sjameel/tt-metal/tt_metal/hw/firmware/src/tt-1xx/ncrisck.cc:17:
[1,0]<stdout>: /home/sjameel/tt-metal/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/dataflow/reader_scatter.cpp: In instantiation of 'void {anonymous}::scatter_along_chunk(const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const uint32_t&, const ScatterReductionType&) [with number_type = short unsigned int; index_type = short unsigned int; uint32_t = long unsigned int]':
[1,0]<stdout>: /home/sjameel/tt-metal/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/dataflow/reader_scatter.cpp:258:72:   required from here
[1,0]<stdout>:   258 |                     scatter_along_chunk<input_std_type, index_std_type>(
[1,0]<stdout>:       |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
[1,0]<stdout>:   259 |                         ctas.input_cb,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~                                  
[1,0]<stdout>:   260 |                         ctas.index_cb,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~                                  
[1,0]<stdout>:   261 |                         ctas.source_cb,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~~                                 
[1,0]<stdout>:   262 |                         ctas.output_cb,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~~                                 
[1,0]<stdout>:   263 |                         ctas.input_stick_size,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~~~~~~~~~                          
[1,0]<stdout>:   264 |                         input_offset,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~                                   
[1,0]<stdout>:   265 |                         input_chunk_length,
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~~~~~~                             
[1,0]<stdout>:   266 |                         index_chunk_length);
[1,0]<stdout>:       |                         ~~~~~~~~~~~~~~~~~~~                             
[1,0]<stdout>: /home/sjameel/tt-metal/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/dataflow/reader_scatter.cpp:135:13: error: cannot convert 'const char*' to 'debug_assert_type_t'
[1,0]<stdout>:   135 |             "Index value {} is bigger than input's dimension size {}.",
[1,0]<stdout>:       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[1,0]<stdout>:       |             |
[1,0]<stdout>:       |             const char*
[1,0]<stdout>: /home/sjameel/tt-metal/tt_metal/hw/inc/debug/assert.h:45:41: note: in definition of macro 'ASSERT'
[1,0]<stdout>:    45 |             assert_and_hang(__LINE__, ##__VA_ARGS__); \
```

--- a/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/dataflow/reader_scatter.cpp
+++ b/ttnn/cpp/ttnn/operations/data_movement/scatter/device/kernels/dataflow/reader_scatter.cpp
@@ -204,11 +204,6 @@ FORCE_INLINE void scatter_along_chunk(
         if (index_value < input_offset || index_value >= input_offset + input_chunk_size) {
             continue;
         }
-        ASSERT(
-            index_value < input_stick_size,
-            "Index value {} is bigger than input's dimension size {}.",
-            index_value,
-            input_stick_size);
         if (index_value >= input_stick_size) {
             continue;
         }

